### ISSUE TITLE
Uncaught Error: Assertion Failed: calling set on destroyed object

### ISSUE THREAD

**Issue by nyon** [offirgolan/ember-cli-nvd3#3]
In my app, I use a search box which filters a table that renders different graphs. In your rendering routine you do not check, whether the component, that gets rendered is already destroyed. As a result the following error is thrown:

```
Uncaught Error: Assertion Failed: calling set on destroyed object
```

I think some of the following code would do the trick in your render routine:

```
if ( !(obj.get('isDestroyed') || obj.get('isDestroying')) ) {
  // do your destroying code setting stuff
}
```

**Comment by offirgolan** [comment]
Can I get a better stack trace of where this issue is coming from?

**Comment by nyon** [comment]
Of course.

```
Uncaught Error: Assertion Failed: calling set on destroyed

object_emberMetalCore.default.assert              @ ember.debug.js:5406
set                                               @ ember.debug.js:20842
exports.default._emberMetalMixin.Mixin.create.set @ ember.debug.js:36650
(anonymous function)                              @ nvd3-chart.js:50       (which is: var svgContainer = d3.select(selector).append("svg"); )
nv.render.renderLoop                              @ nv.d3.js:94            (which is: chart = graph.generate();)
```

I'm using
ember: 1.13.9
ember-cli: 1.13.6
ember-cli-d3: 0.6.0
ember-cli-nvd3: 0.0.4

**Comment by offirgolan** [comment]
@nyon can you checkout master into your application and give it a test?

In package.json do this
`ember-cli-nvd3: 'offirgolan/ember-cli-nvd3#master'`

then npm install and let me know if that fixed it. I wasnt able to recreate your issue but this should do the trick.

**Comment by nyon** [comment]
Thanks, I will give it a try on monday and tell you whether it has worked

**Comment by offirgolan** [comment]
Should be resolved with `v0.0.5`

**Comment by Amiedeep** [comment]
if you are using Ember.Instrumentation namespace for publishing your events and Ember.subscribe to listen to them in your code, when you publish events then it stack all the subscribers and call them even when they are destroyed. so basically, when for example your route is destroyed which is running subscribe statements it will fail and give above error. 

Here is a solution to this problem. Use Ember.Instrumentation.reset() to remove previous subscribers which are destroyed. This solution helped me in acceptance tests where app was destroyed after tests are finished. I wrote Ember.Instrumentation.reset() in startApp function of module-for-acceptance.js and it worked. Hope that helps.
